### PR TITLE
Asset ID

### PR DESCRIPTION
**Pull request by alexeykoren**
- Asset id test vectors
- ZSA value commitment building and test

### CODE REVIEW

**Comment by alexeykoren** [comment]
It is based on a Vivek's branch that is not yet merge so probably its better to merge it first

File: zcash_test_vectors/orchard/commitments.py [hunk 1]
```diff
@@ -86,5 +93,27 @@ def randbytes(l):
 
     assert value_commit(rcv, v) == VALUE_COMMITMENT_RANDOMNESS_BASE * rcv + VALUE_COMMITMENT_VALUE_BASE * v
 
+# Test consistency of ValueCommit^{Orchard} with precomputed generators for non-native asset
+def test_value_commit_zsa():
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/commitments.py:97]: remove `test_value_commit_zsa()` and simply add another assert into `value_commit_zsa()`.

File: zcash_test_vectors/orchard/commitments.py [hunk 2]
```diff
@@ -18,6 +22,9 @@ def homomorphic_pedersen_commitment(rcv: Scalar, D, v: Scalar):
 def value_commit(rcv: Scalar, v: Scalar):
     return homomorphic_pedersen_commitment(rcv, b"z.cash:Orchard-cv", v)
 
+def value_commit_zsa(rcv: Scalar, v: Scalar, asset_id: Point):
+    return asset_id * v + group_hash(b"z.cash:Orchard-cv", b"r") * rcv
+
```
[the remark this answers is not in the document]
Comment (PaulLaux) [reply zcash_test_vectors/orchard/commitments.py:27]: also, should use `asset` not `asset_id`.

[the remark this answers is not in the document]
Comment (alexeykoren) [reply zcash_test_vectors/orchard/commitments.py:27]: done

File: zcash_test_vectors/orchard/asset_id.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    rand = Rand(randbytes)
+
+    test_vectors = []
+    for i in range(0, 20):
+        sk = SpendingKey(rand.b(32))
+        fvk = FullViewingKey.from_spending_key(sk)
+
+        key_bytes = bytes(fvk.ivk())
+        description = ''.join(rng.choice(string.ascii_uppercase + string.digits) for _ in range(512))
+        description_bytes = description.encode("UTF-8")
```
[the remark this answers is not in the document]
Comment (alexeykoren) [reply zcash_test_vectors/orchard/asset_id.py]: done

File: zcash_test_vectors/orchard/note_encryption.py [hunk 4]
```diff
@@ -208,28 +208,29 @@ def randbytes(l):
         g_d = diversify_hash(d)
 
         is_native = i < 10
-        note_type = None if is_native else bytes(Point.rand(rand))
+        asset = None if is_native else bytes(Point.rand(rand))
+        asset_descr = None if is_native else randbytes(512)
```
[the remark this answers is not in the document]
Comment (alexeykoren) [reply zcash_test_vectors/orchard/note_encryption.py]: ok

File: zcash_test_vectors/orchard/commitments.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexeykoren) [reply zcash_test_vectors/orchard/commitments.py]: done

File: zcash_test_vectors/orchard/commitments.py (hunk 1 above)
Comment (alexeykoren) [reply zcash_test_vectors/orchard/commitments.py]: done

File: zcash_test_vectors/orchard/note_encryption.py [hunk 5]
```diff
@@ -208,28 +208,29 @@ def randbytes(l):
...
             rand.u64(),
-            note_type,
+            asset,
             rseed,
             memo
         )
 
         rcv = rcv_trapdoor(rand)
-        cv = value_commit(rcv, Scalar(np.v))
+        cv = value_commit(rcv, Scalar(np.v)) if is_native else value_commit_zsa(rcv, Scalar(np.v), asset_id(asset, asset_descr))
```
[the remark this answers is not in the document]
Comment (alexeykoren) [reply zcash_test_vectors/orchard/note_encryption.py]: removed

File: pyproject.toml [hunk 6]
```diff
@@ -62,3 +62,4 @@ orchard_note_encryption = "zcash_test_vectors.orchard.note_encryption:main"
 orchard_poseidon = "zcash_test_vectors.orchard.poseidon:main"
 orchard_poseidon_hash = "zcash_test_vectors.orchard.poseidon:hash_test_vectors"
 orchard_sinsemilla = "zcash_test_vectors.orchard.sinsemilla:main"
+orchard_asset_id="zcash_test_vectors.orchard.asset_id:main"
```
[the remark this answers is not in the document]
Comment (alexeykoren) [reply pyproject.toml]: done

**Review by PaulLaux — DISMISSED** [review]
looks good. For `/test-vectors/rust` directory - remove all files from this PR except 
`orchard_asset_id.rs`
`orchard_key_components.rs`
`orchard_note_encryption.rs`
(same as in `/test_vectors/json` folder)

All other files include just whitespace changes and we don't want them as part of our diff.

**Comment by PaulLaux** [comment]
conclude by copying the `rs` files into https://github.com/QED-it/orchard/pull/34

**Review by PaulLaux — APPROVED** [review]

**Pull request merged**